src/View/WebView.php

```diff
@@ -79,6 +85,14 @@ class WebView extends View
      */
     private const PH_BODY_END = '<![CDATA[YII-BLOCK-BODY-END]]>';
 
+    /**
+     * The widgets that are currently being rendered (not ended). This property is maintained by {@see beginWidget()}
+     * and {@see static::endWidget()} methods.
+     *
+     * @var Widget[] $stack
+     */
+    protected static $stack;
```

Comment:
Why static?

Reply:
I'm not sure if one instance will always be used when rendering a page from various view files / widgets / templates / layouts, etc. If this is guaranteed, then static can be removed.

Reply:
Multiple instances of view is the main reason to make it non-static. They should not interfere each other - if one view instance do somethings stupid, it should not affect other view instances.

Reply:
> So multiple instances means multiple unrelated documents rendered?

Not only. Even nested views components should not interfere each other. It still will not be perfect, because output buffering stack in PHP is global, but having another static property for state in class will only make it worse.